include/boost/json/detail/value_to.hpp

```diff
@@ -0,0 +1,207 @@
+//
+// Copyright (c) 2019 Vinnie Falco (vinnie.falco@gmail.com)
+// Copyright (c) 2020 Krystian Stasiowski (sdkrystian@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/vinniefalco/json
+//
+
+#ifndef BOOST_JSON_DETAIL_VALUE_TO_HPP
+#define BOOST_JSON_DETAIL_VALUE_TO_HPP
+
+#include <boost/json/number_cast.hpp>
+#include <boost/json/value.hpp>
+#include <boost/json/detail/value_conversion.hpp>
+
+#include <type_traits>
+
+namespace boost {
+namespace json {
+
+template<class>
+struct value_to_tag { };
```

Comment:
We have a documentation problem here, because `value_to_tag` needs to be in the documentation as a linkable item, and it needs an entry in the quick reference. So it needs to be in a public header (i.e. not a header in detail). You will need to move it and figure this out. Same for `value_from_tag`.